sonar-plugin-api/src/main/java/org/sonar/api/batch/bootstrap/ProjectDefinition.java

```diff
@@ -351,6 +353,10 @@ private void setParent(ProjectDefinition parent) {
     return result;
   }
 
+  public boolean shouldAnalyzeParentModules() {
```

Comment:
This is also an addition to the API. You should move that logic out of the API (could be a static helper method in ProjectReactorBuilder or FileIndexer)